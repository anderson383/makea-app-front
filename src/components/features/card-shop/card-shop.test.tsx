
import CardShop from '.'
import { act } from 'react-dom/test-utils';
import {render, fireEvent, waitFor} from '@testing-library/react-native'
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';


const navigate = jest.fn()
const moduleRouter = {
  useNavigate: () => navigate
}
jest.mock('react-router-native', () => moduleRouter)

const mockProduct = {
  id: 'idtest',
  name: 'Product test',
  description: '',
  image: '',
  price: 1000,
  categoria: 'categoriatest',
  isAddCart: true
}

const mockStore = configureStore([]);

describe('Test feature CardShop product', () => {
  
  const setup = async (store:any) => {
    return await waitFor(() => render(
      <Provider store={store}>
        <CardShop />
      </Provider>
    )) 
  }

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test(`Should render component`, async () => {
    const store = mockStore({
      product: {
        products: [mockProduct]
      }
    });
    await act(async () => {
      const {toJSON} = await setup(store)

      expect(toJSON()).toMatchSnapshot()
    });
  })

  test(`Should click at buy button`, async () => {
    await act(async () => {
      const store = mockStore({
        product: {
          products: [mockProduct]
        }
      });
      const {getByTestId} = await setup(store)

      const button = getByTestId('buttonBuy')
      fireEvent.press(button)
      expect(store.getActions()[0].payload).toEqual([])
      expect(navigate).toHaveBeenCalledWith('/success-pay')
    });
  })
})