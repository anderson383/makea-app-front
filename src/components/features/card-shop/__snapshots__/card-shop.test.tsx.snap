// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test feature CardShop product Should render component 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingTop": 54,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 73,
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 35,
          "fontWeight": "bold",
        }
      }
    >
      Carrito
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={20}
      vbWidth={20}
      width={20}
      xml=" 
<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.2678 12.8521C14.5265 11.4896 15.3117 9.66507 15.3117 7.65584C15.3117 3.44108 11.8821 0 7.65584 0C3.42954 0 0 3.44108 0 7.65584C0 11.8706 3.42954 15.3117 7.65584 15.3117C9.3764 15.3117 10.9584 14.7459 12.2401 13.7874L17.9329 19.4802C18.0715 19.6188 18.2447 19.6881 18.4179 19.6881C18.5911 19.6881 18.7758 19.6188 18.9029 19.4802C19.1685 19.2146 19.1685 18.7759 18.9029 18.4987L13.2678 12.8521ZM1.38567 7.65584C1.38567 4.2032 4.19166 1.38567 7.65584 1.38567C11.1085 1.38567 13.926 4.19166 13.926 7.65584C13.926 11.12 11.12 13.926 7.65584 13.926C4.19166 13.926 1.38567 11.1085 1.38567 7.65584Z" fill="#212224"/>
</svg>

"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M13.2678 12.8521C14.5265 11.4896 15.3117 9.66507 15.3117 7.65584C15.3117 3.44108 11.8821 0 7.65584 0C3.42954 0 0 3.44108 0 7.65584C0 11.8706 3.42954 15.3117 7.65584 15.3117C9.3764 15.3117 10.9584 14.7459 12.2401 13.7874L17.9329 19.4802C18.0715 19.6188 18.2447 19.6881 18.4179 19.6881C18.5911 19.6881 18.7758 19.6188 18.9029 19.4802C19.1685 19.2146 19.1685 18.7759 18.9029 18.4987L13.2678 12.8521ZM1.38567 7.65584C1.38567 4.2032 4.19166 1.38567 7.65584 1.38567C11.1085 1.38567 13.926 4.19166 13.926 7.65584C13.926 11.12 11.12 13.926 7.65584 13.926C4.19166 13.926 1.38567 11.1085 1.38567 7.65584Z"
          fill={
            {
              "payload": 4280361508,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <RCTScrollView
    style={
      {
        "backgroundColor": "#E5E5E5",
        "padding": 16,
      }
    }
  >
    <View>
      <View
        style={
          {
            "gap": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "flexDirection": "row",
              "gap": 16,
              "overflow": "hidden",
              "padding": 16,
              "position": "relative",
            }
          }
        >
          <View
            style={
              {
                "position": "absolute",
                "right": 16,
                "top": 16,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="deleteItem"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="11"
                bbWidth="11"
                fill="none"
                focusable={false}
                height={11}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 11,
                      "width": 11,
                    },
                  ]
                }
                vbHeight={11}
                vbWidth={11}
                width={11}
                xml="<svg width="11" height="11" viewBox="0 0 11 11" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M11 1.1L9.9 0L5.5 4.4L1.1 0L0 1.1L4.4 5.5L0 9.9L1.1 11L5.5 6.6L9.9 11L11 9.9L6.6 5.5L11 1.1Z" fill="#D6D6D6"/>
</svg>
"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    clipRule={0}
                    d="M11 1.1L9.9 0L5.5 4.4L1.1 0L0 1.1L4.4 5.5L0 9.9L1.1 11L5.5 6.6L9.9 11L11 9.9L6.6 5.5L11 1.1Z"
                    fill={
                      {
                        "payload": 4292269782,
                        "type": 0,
                      }
                    }
                    fillRule={0}
                    propList={
                      [
                        "fill",
                        "fillRule",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <Image
            source={
              {
                "height": 87,
                "uri": "",
                "width": 87,
              }
            }
            style={
              {
                "borderRadius": 8,
                "overflow": "hidden",
              }
            }
          />
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <Text
              numberOfLines={2}
              style={
                {
                  "fontWeight": "bold",
                }
              }
            >
              Product test
            </Text>
            <Text
              numberOfLines={2}
            />
          </View>
          <Text
            style={
              {
                "color": "#FE2550",
                "fontWeight": "bold",
              }
            }
          >
            $ 1,000
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginTop": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "gap": 8,
            }
          }
        >
          <Text>
            Total:
          </Text>
          <Text
            style={
              {
                "color": "#FE2550",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            $ 1,000
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FE2550",
                "borderRadius": 8,
                "justifyContent": "center",
                "minHeight": 62,
                "opacity": 1,
                "paddingHorizontal": 16,
              }
            }
            testID="buttonBuy"
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontWeight": "bold",
                }
              }
            >
              Pagar ahora
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
